PhysicsTools/NanoAOD/python/protons_cff.py

```diff
@@ -0,0 +1,74 @@
+import FWCore.ParameterSet.Config as cms
+from PhysicsTools.NanoAOD.common_cff import *
+from Configuration.Eras.Modifier_run2_miniAOD_80XLegacy_cff import run2_miniAOD_80XLegacy
+from Configuration.Eras.Modifier_run2_nanoAOD_94XMiniAODv1_cff import run2_nanoAOD_94XMiniAODv1
+from Configuration.Eras.Modifier_run2_nanoAOD_94XMiniAODv2_cff import run2_nanoAOD_94XMiniAODv2
+from Configuration.Eras.Modifier_run2_nanoAOD_94X2016_cff import run2_nanoAOD_94X2016
+from Configuration.Eras.Modifier_run2_nanoAOD_102Xv1_cff import run2_nanoAOD_102Xv1
+
+protonTable = cms.EDProducer("ProtonProducer",
+                             tagRecoProtonsSingle = cms.InputTag("ctppsProtons", "singleRP"),
+                             tagRecoProtonsMulti  = cms.InputTag("ctppsProtons", "multiRP"),
+                             tagTrackLite         = cms.InputTag("ctppsLocalTrackLiteProducer")
+)
+
+singleRPTable = cms.EDProducer("SimpleProtonTrackFlatTableProducer",
+    src = cms.InputTag("ctppsProtons","singleRP"),
+    cut = cms.string(""),
+    name = cms.string("Proton_singleRP"),
+    doc  = cms.string("bon"),
+    singleton = cms.bool(False),
+    extension = cms.bool(False),
+    skipNonExistingSrc = cms.bool(True),
+    variables = cms.PSet(
+        xi = Var("xi",float,doc="xi or dp/p",precision=12),
+        xiUnc = Var("xiError",float,doc="uncertainty on xi or dp/p",precision=8),
+        thetaY = Var("thetaY",float,doc="th y",precision=13),
+        thetaYUnc = Var("thetaYError",float,doc="theta Y uncertainty",precision=8),
+    ),
+    externalVariables = cms.PSet(
+        decRPId = ExtVar("protonTable:protonRPId",int,doc="Detector ID",precision=8), 
+        sector45 = ExtVar("protonTable:singleRPsector45",bool,doc="LHC sector 45"),
+        sector56 = ExtVar("protonTable:singleRPsector56",bool,doc="LHC sector 56"),
+    ),
+)
+
+multiRPTable = cms.EDProducer("SimpleProtonTrackFlatTableProducer",
+    src = cms.InputTag("ctppsProtons","multiRP"),
+    cut = cms.string(""),
+    name = cms.string("Proton_multiRP"),
+    doc  = cms.string("bon"),
+    singleton = cms.bool(False),
+    extension = cms.bool(False),
+    skipNonExistingSrc = cms.bool(True),
+    variables = cms.PSet(
+        xi = Var("xi",float,doc="xi or dp/p",precision=12),
+        xiUnc = Var("xiError",float,doc="uncertainty on xi or dp/p",precision=8),
+        vtxY = Var("vy()",float,doc="vertex y",precision=11),
+        vtxYUnc = Var("vyError",float,doc="vy Error",precision=8),
+        pt = Var("pt",float,doc="pt",precision=15),
+        thetaX = Var("thetaX",float,doc="theta x",precision=13),
+        thetaXUnc = Var("thetaXError",float,doc="theta X uncertainty",precision=8),
+        thetaY = Var("thetaY",float,doc="theta y",precision=13),
+        thetaYUnc = Var("thetaYError",float,doc="theta Y uncertainty",precision=8),
+        chi2 = Var("chi2",float,doc="chi 2",precision=15),
+        ndof = Var("ndof()",int, doc="n dof", precision=8),
```

Comment:
can you replace with the normalised chi2 ? 
as done with the protonTable
ppsTab->addColumn<float>("normalizedChi2", normalizedChi2, "chi2 over NDF", 8);

Reply:
There actually is a reason for storing chi2 and not normalised chi2: with the default configuration, the multi-RP fit has 0 degrees of freedom and thus the normalised chi2 would be NAN - not very useful.

Reply:
ok, 0 ndof means the output is equal to the input ?

one can store
Var("?(ndof>0)?(chi2/nodf):chi2, float,doc="",precision=8)

Reply:
> ok, 0 ndof means the output is equal to the input ?

Not quite. ndof=0 means that the number of fit parameters (for us xi, theta*_x, theta*_y and y*) is equal to the number of constraints (for us x and y measurements in 2 RP detectors). In this mode, the chi2 minimisation effectively finds a solution to 4 non-linear equations. The value of (residual) chi2 should be very small (only given by Minuit precision goal). A large number means fit failure - that's why we like to store it.

> one can store
> Var("?(ndof>0)?(chi2/nodf):chi2, float,doc="chi2 over NDF",precision=8)

Yes, this is a possibility. Personally, I'd prefer each variable to have unique meaning. However, if you deem this saving necessary, fine with me.

Reply:
there is also the bool validFit = Var("validFit",bool,doc="valid Fit") that might be related to chi2 and ndof
the suggestion is to change into something that people apply in data analysis.

For POG investigations one always need to go back to the miniAOD.

Reply:
> there is also the bool validFit = Var("validFit",bool,doc="valid Fit") that might be related to chi2 and ndof
> the suggestion is to change into something that people apply in data analysis.

`validFit` only contains the status returned by Minuit. This is actually used in the analysis.

> For POG investigations one always need to go back to the miniAOD.

I see.

I see that we might need to reshape our proposal a little. However, I am afraid about the timeline - re-discussing the proposal will take time. Can we go in steps: approve this PR and then do improve things later? If no, what is the acceptance criterion? E.g. are the event sizes reported by @jwill24 tolerable? Or too large by a factor of 2, ... ?